esiclient/tests/functional/v1/test_node.py

```diff
@@ -0,0 +1,111 @@
+#    Copyright (c) 2016 Mirantis, Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import ddt
+from tempest.lib import exceptions
+
+from esiclient.tests.functional import base
+from esiclient.tests.functional import utils
+
+
+@ddt.ddt
+class NodePowerTests(base.ESIBaseTestClass):
+    """Functional tests for esi node power commands."""
+
+    @classmethod
+    def setUpClass(cls):
+        super(NodePowerTests, cls).setUpClass()
+        cls._init_dummy_project(cls, 'owner_proj', 'owner')
+        cls._init_dummy_project(cls, 'owner_proj_2', 'owner')
+        cls._init_dummy_project(cls, 'lessee_proj', 'lessee')
+        cls._init_dummy_project(cls, 'lessee_proj_2', 'lessee')
+
+    def setUp(self):
+        super(NodePowerTests, self).setUp()
+        self.clients = NodePowerTests.clients
+        self.users = NodePowerTests.users
+        self.projects = NodePowerTests.projects
+
+    def test_owner_node_power_on_off(self):
+
+        node = utils.node_create(self.clients['admin'],
+                                 name="owned")
+        self.addCleanup(utils.node_delete,
+                        self.clients['admin'],
+                        node['name'])
+
+        utils.node_set(self.clients['admin'],
+                       identifier=node['name'],
+                       param='owner',
+                       value=self.projects['owner_proj']['id'])
+
+        utils.node_power_on(self.clients['owner_proj-owner'], node['name'])
+        utils.node_power_off(self.clients['owner_proj-owner'], node['name'])
+
+        self.assertRaises(exceptions.CommandFailed,
+                          utils.node_power_on,
```

Comment:
This should probably be split off into another test (a negative test).

Reply:
I had thought of doing it like that, but for speed's sake I had went against it. Having to create and delete fewer nodes would be nice. But splitting it to negative tests would line up with what we have in the spreadsheet more I suppose